synapse/config/appservice.py

```diff
@@ -33,17 +33,21 @@ class AppServiceConfig(Config):
 
     def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         self.app_service_config_files = config.get("app_service_config_files", [])
+        if not isinstance(self.app_service_config_files, list):
```

Comment:
Would be tempted to stricten it by checking everything within is a string, for example:
```suggestion
        if not isinstance(self.app_service_config_files, list) or not all(type(x) is str for x in self.app_service_config_files):
```

Reply:
Good shout!